src/library/dlgtagfetcher.h

```diff
@@ -58,4 +63,6 @@ class DlgTagFetcher : public QDialog,  public Ui::DlgTagFetcher {
         UnknownError,
     };
     NetworkResult m_networkResult;
+    const TrackModel* m_pTrackModel;
```

Comment:
Okay. I'll leave a TODO comment in `DlgTrackInfo` and `DlgTagFetcher` classes.
By the way, what should be a better solution that is not a hack? We need the classes to be configurable based on whether they can switch tracks or not.

Reply:
Coming up with a new design is out of the scope of this review. I'm not able to provide this ad-hoc without diving into the details. Time is limited.

Reply:
I think a new design would require many signals back and forth between WTrackTableView and WTrackMenu so that WTrackMenu never works with TrackModel or QModelIndex, only TrackPointers.